src/huggingface_hub/inference/_client.py

```diff
@@ -1305,3 +1306,50 @@ def _resolve_url(self, model: Optional[str] = None, task: Optional[str] = None)
             # Otherwise, we use the default endpoint
             else f"{INFERENCE_ENDPOINT}/models/{model}"
         )
+
+    def get_model_status(self, model: Optional[str] = None, *, token: Optional[str] = None) -> ModelStatus:
+        """
+            A function which returns the status of a specific model, from the Inference API.
+
+            Args:
+                model (`str`, *optional*):
+                    Identifier of the model for witch the status gonna be checked. If model is not provided, 
+                    the model associated with this instance of [`InferenceClient`] will be used. Only InferenceAPI service can be checked so the 
+                    identifier cannot be a URL.
+
+                token (`str`, *optional*)
+                      Hugging Face token. Will default to the locally saved token. Pass `token=False` if you don't want to send
+            your token to the server.
+            
+
+            Returns:
+                [`ModelStatus`]: An instance of ModelStatus dataclass, containing information,
+                             about the state of the model: load, state, compute type and framework.
+                      
+            Raises:
+                [`ValueError`]:
+                    If the model is missing, meaning is not provided. 
+                    And if if the API returns an error message(missing model).
+                [`NotImplementedError`]:
+                    If the provided model is a URL.
+                
+        """
+        model = model or self.model
+        if model is None:
+             raise ValueError("Model id not provided")
+        if model.startswith("https://"):
+            raise NotImplementedError("Model status is only available for Inference API endpoints.")
```

Comment:
How to get status for Inference API endpoinits?

Reply:
Nvm found work in another related ticket. https://github.com/huggingface/huggingface_hub/issues/1541